### PR TITLE
LINT: Initial changes for Lint warnings removal

### PR DESCRIPTION
**Pull request by umerimran-10xe**
- [x] Fixes MODDUP warnings
- [x] Fixes Verilator arguments usage
- [x] Fixes CASEINCOMPLETE warnings
- [x] Fixes some WIDTHEXPAND and WIDTHCONCAT warnings in packages
- [x] Fixes some BLKSEQ warnings
- [x] Adds verilator lint off configuration file

### CODE REVIEW

**Review by JeanRochCoulon — APPROVED** [review]
Great modification of code. I approve, awaiting for the CI pass.

File: Makefile [hunk 1]
```diff
@@ -160,50 +160,37 @@ src :=  corev_apu/tb/axi_adapter.sv
...
+        vendor/pulp-platform/tech_cells_generic/src/deprecated/pulp_clk_cells.sv     \
+        vendor/pulp-platform/tech_cells_generic/src/rtl/tc_clk.sv                    \
         corev_apu/tb/ariane_testharness.sv                                           \
         corev_apu/tb/ariane_peripherals.sv                                           \
         corev_apu/tb/rvfi_tracer.sv                                                  \
         corev_apu/tb/common/uart.sv                                                  \
         corev_apu/tb/common/SimDTM.sv                                                \
         corev_apu/tb/common/SimJTAG.sv
 
-# SV32 MMU for CV32, SV39 MMU for CV64
```
Comment (mustafa7755) [inline Makefile:200]: Why are these removed? Intentional or accidental?

File: corev_apu/tb/ariane_soc_pkg.sv [hunk 2]
```diff
@@ -70,26 +70,26 @@ package ariane_soc;
...
+    BTBEntries: int'(cva6_config_pkg::CVA6ConfigBTBEntries),
+    BHTEntries: int'(cva6_config_pkg::CVA6ConfigBHTEntries),
     // idempotent region
-    NrNonIdempotentRules:  1,
-    NonIdempotentAddrBase: {64'b0},
-    NonIdempotentLength:   {DRAMBase},
-    NrExecuteRegionRules:  3,
-    ExecuteRegionAddrBase: {DRAMBase,   ROMBase,   DebugBase},
-    ExecuteRegionLength:   {DRAMLength, ROMLength, DebugLength},
+    NrNonIdempotentRules:  unsigned'(1'b0),
```
[the remark this answers is not in the document]
Comment (umerimran-10xe) [reply corev_apu/tb/ariane_soc_pkg.sv:77]: That's a typo, I have changed it in the local, let me run the CI checks here on the local machine before pushing changes

File: Makefile (hunk 1 above)
Comment (umerimran-10xe) [reply Makefile:200]: Intentional, these were module duplication in the Makefile and CVA6 Flist

**Comment by umerimran-10xe** [comment]
There should be a linter setup of trimming trailing whitespaces for the whole repository so that it will not be introduced in future PRs.

**Comment by JeanRochCoulon** [comment]
@umerimran-10xe    The Thales CI is failed, I think it is due to the fact that rebase is not done, can you rebase the PR please ?

**Comment by umerimran-10xe** [comment]
Since the Thales CI is failing, let me review the PR once again and maybe divide this PR into two PRs for easy review

**Comment by umerimran-10xe** [comment]
Ran the compliance tests locally, they are passing now, @JeanRochCoulon to review the changes once the CI passes. I have changed the default config in the Makefile target to be `CV32A60X` since that's the one we are more focused on, is it okay or should I revert that change ?

**Comment by JeanRochCoulon** [comment]
@umerimran-10xe    You need to revert the Makefile change, otherwise the Github actions will run in 32 bits mode which is not wanted. Maybe it does not work in this configuration.

**Comment by JeanRochCoulon** [comment]
Last error, here is the log

image

**Comment by umerimran-10xe** [comment]
Sure, let me see that and add the missing file in the Makefile.

**Comment by JeanRochCoulon** [comment]
Unfornately the vcs-testharness smoke_tests do not pass... If you do not have VCS, tell me I can try to fix it.

**Comment by umerimran-10xe** [comment]
Are these tests in Thales CI, it seems the CI is passing. Currently we don't use VCS, please go ahead with fixing the required issue here with the `vcs-testharness`
https://riscv-ci.pages.thales-invia.fr/dashboard/dashboard_cva6_0.html

**Comment by JeanRochCoulon** [comment]
I will have a look. Can you give me the rights to propose a PR to your PR (it will be easier to collaborate on the PR) ?

**Comment by umerimran-10xe** [comment]
Sure, I think it's done now, can you please check that. Also since github issues is down for a little bit, maybe we will observe degraded performance for some time until the issue is resolved. 
https://www.githubstatus.com/
